include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_fpga.h

```diff
@@ -39,6 +39,14 @@ namespace dpl
 {
 namespace __par_backend_hetero
 {
+template <typename _ExecutionPolicy>
+auto
+__device_policy(_ExecutionPolicy&& __exec)
```

Comment:
Do we really need to take `ExecutionPolicy` by forwarding reference here? 
The only action we do is reading the `queue` and it is never depends on the policy value category.
So maybe we can take it as const lvalue reference and avoid forwarding the policy while calling the function? What do you think?

Reply:
Looks like we should create new policy instance with `__kernel_name` from template param policy class.
I think it's the main idea.

Reply:
Yes, we need to create a new policy with the same kernel name and with the same queue. As far as I understand, forwarding reference is redundant here.

Also, in all places where this function is called we have a comment:

`// workaround until we implement more performant version for patterns`

If it relates to the policy re-creation with the same kernel name and queue, I guess the comment should be moved inside the new function.